Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -882,6 +882,51 @@ public static TTo As<TTo>(this object subject)
             return (await task).WithInnerException<TInnerException>(because, becauseArgs);
         }
 
+        /// <summary>
+        /// Asserts that the thrown exception has a parameter which name matches <paramref name="paramName" />.
+        /// </summary>
+        /// <param name="parent">The <see cref="ExceptionAssertions{TException}"/> containing the thrown exception.</param>
+        /// <param name="paramName">The expected name of the parameter</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static ExceptionAssertions<TException> WithParameterName<TException>(
+            this ExceptionAssertions<TException> parent,
+            string paramName,
```

Comment:
I know this has already been merged, but I'd suggest adding argument validation (!= null) for `paramName` here to improve the stacktrace in case people misuse the method.

Reply:
I didn't think about that case, but `ArgumentException` allows a `null` `ParamName`.

Maybe I'm misunderstanding you, can you give an example on "misuse"?